.github/workflows/llvm.yml

```diff
@@ -38,19 +33,30 @@ jobs:
         uses: actions/cache@v4
         with:
           path: ./llvm
-          key: llvm-${{ matrix.llvm_version }}
+          key: llvm-libs-${{ matrix.llvm_version }}-${{ hashFiles('.github/workflows/llvm.yml') }}-linux
```

Comment:
**issue:** The hash on this workflow file triggers a rebuild every time we change anything here. This includes every time we bump the version for `install-crystal` action after each release. I think that's quite frequent in relation to the long build times for LLVM.
I'm wondering if we could narrow this down somehow, maybe call a separate workflow which handles only building LLVM?